impl/maven-jline/src/main/java/org/apache/maven/jline/LegacyPlexusInteractivity.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.jline;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.maven.api.annotations.Experimental;
+import org.codehaus.plexus.components.interactivity.InputHandler;
+import org.codehaus.plexus.components.interactivity.OutputHandler;
+import org.codehaus.plexus.components.interactivity.Prompter;
+import org.codehaus.plexus.components.interactivity.PrompterException;
+import org.eclipse.sisu.Priority;
+
+/**
+ * This class is injected into any legacy component that would want to use legacy "Plexus Interactivity API".
+ * It simply delegates to Maven4 API {@link DefaultPrompter}.
+ */
+@Experimental
+@Named
+@Singleton
+@Priority(10)
```

Comment:
@cstamas Would it make sense to move that one to `compat/maven-compat` ? The idea would be to avoid introducing any new dependency on sisu in `impl/*` jars ?